azurelinuxagent/common/future.py

```diff
@@ -1,5 +1,9 @@
+import platform
 import sys
 
+if float(sys.version[:3]) >= 3.5:
```

Comment:
Function is deprecated in 3.5, I think we should use that version. Anyway on SUSE in 3.6 the return frrom the platform module is already broken, i.e. it returns an empty strings as in ('','','') so if I set this to 3.7 that does not help me at all.

Reply:
In that case we will have to have a distro specific abstraction, since this change would break Ubuntu.

Reply:
Problem with "distro specific abstraction" is that we cannot detect the distribution it's a chicken and egg problem.

But I am not certain about why this would break other distros:
- If the user installs from source with pip then pip will pull in the distro package if python version > 3.5
- If other distros create a new package for the new version of WALinuxAgent then they need to also create a package for distro and release this together
- If the user installs from source using setup.py then the user gets an error if distro is not installed.

I do not see how making this retroactive to 3.5 breaks anything.

It is possible to not check the version at all and just rely on exception handling and value checking. However since the changes are ties to a specific Python version I think having the version checks makes things more explicit.

Reply:
Self-update is the other channel, which will break.